test/Game/testERC721.test.js

```diff
@@ -12,52 +12,103 @@ async function getRandom() {
 };
+*/
 
 const erc721Tests = require('../erc721')(
-  withSnapshot(['ChildGameToken'], async () => {
-    const {gameTokenAdmin} = await getNamedAccounts();
-    const others = await getUnnamedAccounts();
-
-    const contract = await ethers.getContract('ChildGameToken');
-
-    const gameAsAdmin = await contract.connect(
-      ethers.provider.getSigner(gameTokenAdmin)
-    );
-    await gameAsAdmin.changeMinter(gameTokenAdmin);
-
-    async function mint(to) {
-      const assets = await supplyAssets(to, [1]);
-      const randomId = await getRandom();
-
-      const receipt = await gameAsAdmin.createGame(
-        to,
-        to,
-        {
-          ...creation,
-          assetIdsToAdd: assets,
-          exactNumOfLandsRequired: 1,
-        },
-        ethers.constants.AddressZero,
-        randomId
-      );
-      const updateEvent = await expectEventWithArgs(
-        contract,
-        receipt,
-        'GameTokenUpdated'
+  withSnapshot(
+    ['GameAsset1155', 'GameAsset721', 'ChildGameToken'],
+    async () => {
+      const {gameTokenAdmin} = await getNamedAccounts();
+      const others = await getUnnamedAccounts();
+
+      const contract = await ethers.getContract('ChildGameToken');
+
+      const gameAsAdmin = await contract.connect(
+        ethers.provider.getSigner(gameTokenAdmin)
       );
-      const tokenId = BigNumber.from(updateEvent.args[1]);
-      return {receipt, tokenId};
-    }
+      await gameAsAdmin.changeMinter(gameTokenAdmin);
+
+      async function mint(to) {
+        const assets = await supplyAssets(to, [1]);
+        const randomId = await getRandom();
 
-    return {contractAddress: contract.address, users: others, mint};
-  }),
+        const receipt = await gameAsAdmin.createGame(
+          to,
+          to,
+          newCreationAdd1155(creation, assets),
+          ethers.constants.AddressZero,
+          randomId
+        );
+        const updateEvent = await expectEventWithArgs(
+          contract,
+          receipt,
+          'GameTokenUpdated'
+        );
+        const tokenId = BigNumber.from(updateEvent.args[1]);
+        return {receipt, tokenId};
+      }
+
+      async function grant(contract, users) {
+        const assetContract1155 = await ethers.getContract('GameAsset1155');
+        const assetContract721 = await ethers.getContract('GameAsset721');
+
+        await [...users].map(async (user) => {
+          const assetContract1155AsUser = await assetContract1155.connect(
+            ethers.provider.getSigner(user)
+          );
+          await assetContract1155AsUser.setApprovalForAll(
+            contract.address,
+            true
+          );
+          const assetContract721AsUser = await assetContract721.connect(
+            ethers.provider.getSigner(user)
+          );
+          await assetContract721AsUser.setApprovalForAll(
+            contract.address,
+            true
+          );
+        });
+      }
```

Comment:
Why not do the approve directly in the mint function so that we don't have to add this grant function